### ISSUE TITLE
Stop auto-raise on focus

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jimradford/superputty#361]
```
This is rather specialist, but I have the Windows 7 registry hack 
(http://sinewalker.wordpress.com/2010/03/10/ms-windows-focus-follows-mouse-regis
try-hacks/) to focus windows on mouseover, but *not* raise them. This works 
fine on every other application, bar superputty which raises the window to the 
foreground :(

Is there any chance this can be fixed, please? Being that this application is 
likely used by people familiar with *nix X-windows, I'm sure there are others 
who would appreciate this. Thanks!
```

Original issue reported on code.google.com by `phw...@gmail.com` on 21 Aug 2013 at 9:55

**Comment by yooden** [comment]
Same here - annoying behaviour. Please make the pain stop.

**Comment by nimrodl** [comment]
This is still an issue after the long awaited 1.4.0.9 release a month ago.

**Comment by jimradford** [comment]
@nimrodl  Are you only seeing this behavior with the registry hack referenced earlier or can you provide steps to reproduce without that hack?

**Comment by nimrodl** [comment]
@jimradford Yes, I do have the modifications in place for focus to follow the mouse. It's been a while since I put it in, so I don't recall exactly what mechanism I used to do it, but I'm pretty sure that they all result in the same registry change.  

In any case the correct behavior should not be for the window to raise to the top when it gets focus unless it's clicked on.

**Comment by jimradford** [comment]
Thanks for the feedback. I **am** inclined to merge this feature request in if someone creates and submits a pull request, however I do not plan on spending any time right now trying to implement a workaround myself since I don't imagine it affects a large portion of the SuperPuTTY userbase nor do I consider it a bug due to the fact the default and common windows behavior is being changed. i.e. it works as expected for most installations.

**Comment by nimrodl** [comment]
I understand your position. 

The piece I don't understand is why this application behaves this way and virtually no others do. This type of programming is not my area, but I'll do some googling and see what information I can find. I am also happy to test any builds as needed. 

Thanks for creating this useful tool and continuing to support it.

**Comment by nimrodl** [comment]
So I've been playing around a little knowing that I don't know what I'm doing, but here's what I've come up with. I can prevent the autoraise on focus behavior on the main superputty window with the by removing the following line 
```
diff --git a/SuperPutty/Utils/ChildWindowFocusHelper.cs b/SuperPutty/Utils/ChildWindowFocusHelper.cs
index 88ea3b5..b45c240 100644
--- a/SuperPutty/Utils/ChildWindowFocusHelper.cs
+++ b/SuperPutty/Utils/ChildWindowFocusHelper.cs
@@ -136,7 +136,7 @@ namespace SuperPutty.Utils
                                 else if (m_externalWindow)
                                 {
                                     m_externalWindow = false;
-                                    NativeMethods.BringWindowToTop(this.MainForm.Handle);
+                                    //NativeMethods.BringWindowToTop(this.MainForm.Handle);
                                     this.MainForm.FocusActiveDocument("SHELLHOOK");
                                     //return false;
                                 }


```
There's still a problem with how the putty windows get focused. If the focus passes from another window to superputty, no problem as above. But if the focus passes from another window to putty inside superputty, superputty gets raised to the top and I can't see why. I think it's something to do with the fixes for this issue https://github.com/jimradford/superputty/issues/1 since if I bypass that logic, superputty no longer autoraises to the top, but also no longer raises on click in a putty window.

At this point, I have probably exceeded my abilities, but I'll keep fiddling as time permits and maybe someone who knows more than I do can help out.
